.github/workflows/docker.yml

```diff
@@ -8,18 +8,38 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@master
-      - name: Publish to Github Docker Package Registry
-        uses: elgohr/Publish-Docker-Github-Action@2.21
+      - name: Get the version
+        id: get_version
+        run: |
+          VERSION=${GITHUB_REF#refs/tags/}
+          if [[ ${GITHUB_REF} == "refs/heads/master" ]]; then
+            VERSION=latest
+          fi
+          echo ::set-output name=VERSION::${VERSION}
+      - name: Login ghcr.io
+        uses: docker/login-action@v1
         with:
-          name: oam-dev/kubevela/vela-core
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-          registry: docker.pkg.github.com
-          tags: "latest"
-      - name: Publish to Docker Hub Registry
-        uses: elgohr/Publish-Docker-Github-Action@2.21
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+      - name: Login docker.io
+        uses: docker/login-action@v1
         with:
-          name: oamdev/vela-core
+          registry: docker.io
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-          tags: "latest"
\ No newline at end of file
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+      - uses: docker/build-push-action@v2
+        name: Build & Pushing
+        with:
+          context: .
+          file: Dockerfile
+          labels: |-
+            org.opencontainers.image.source=https://github.com/${{ github.repository }}
+            org.opencontainers.image.revision=${{ github.sha }}
+          platforms: linux/amd64,linux/arm64
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: |-
+            ghcr.io/${{ github.repository }}/vela-core:${{ steps.get_version.outputs.VERSION }}
+            docker.io/oamdev/vela-core:${{ steps.get_version.outputs.VERSION }}
```

Comment:
I noticed two one more tag is added, will both of them be push to the Container Registry respectively? Which one will be used for end-users?

Reply:
both works. it will push to multi registries.

Reply:
one more thing, images pushing to ghcr.io will be private by default. 

need to make images public here. https://github.com/orgs/oam-dev/packages

Reply:
Got it, thanks. I prefer to pushing to only one registry for quick release build as normally `docker push` might be the same time-consuming step. BTW, can those two push be asynchronous?

Reply:
synchronous. one after one in tag list. 
so the behavior will be same as old, push to github cr, push to docker.io

asynchronous mode in early version of buildx but will broken tags in same container registry like `oamdev/vela-core:1 oamdev/vela-core:1.1`. they removed it.

Reply:
We could remove one of them when updates `release.yml`

Reply:
I think keep both of them is OK, this github action  is  asynchronous with other action such as `e2e test`.